java-client/src/test/java11/org/opensearch/client/opensearch/integTest/AbstractGenericClientIT.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.client.opensearch.integTest;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import org.junit.Test;
+import org.opensearch.client.json.JsonpDeserializer;
+import org.opensearch.client.opensearch.core.SearchResponse;
+import org.opensearch.client.opensearch.generic.Bodies;
+import org.opensearch.client.opensearch.generic.Requests;
+import org.opensearch.client.opensearch.generic.Response;
+import org.opensearch.client.opensearch.indices.CreateIndexResponse;
+
+public abstract class AbstractGenericClientIT extends OpenSearchJavaClientTestCase {
+
+    @Test
+    public void shouldReturnSearchResults() throws Exception {
+        final String index = "search_request";
+        createIndex(index);
+
+        try (
+            Response response = javaClient().generic()
+                .execute(
+                    Requests.builder(javaClient())
+                        .endpoint("/" + index + "/_search")
+                        .method("POST")
+                        .json(
+                            "{"
+                                + "    \"sort\": ["
+                                + "        {"
+                                + "            \"name\": \"asc\""
+                                + "        }"
+                                + "    ],"
+                                + "    \"fields\": ["
+                                + "        \"name\""
+                                + "    ],"
+                                + "    \"query\": {"
+                                + "        \"bool\": {"
+                                + "            \"filter\": ["
+                                + "                {"
+                                + "                    \"term\": {"
+                                + "                        \"size\": \"huge\""
+                                + "                    }"
+                                + "                }"
+                                + "            ]"
+                                + "        }"
+                                + "    },"
+                                + "    \"_source\": true\n"
+                                + "}"
+                        )
+                        .build()
+                )
+        ) {
+            assertThat(response.getStatus(), equalTo(200));
+            assertThat(response.getBody().isPresent(), equalTo(true));
+
+            final SearchResponse<?> r = Bodies.json(
+                response.getBody().get(),
+                SearchResponse.createSearchResponseDeserializer(JsonpDeserializer.of(ShopItem.class)),
+                javaClient()
+            );
```

Comment:
Pretty funny looking, especially unclear why it needs the client. I think I'd want `response.getBody().json()`

Reply:
I wish I have a better way, but what we need is actually `JsonpMapper` instance which we could:
 - ask directly
 - ask from `Transport`
 - ask from `OpenSearchClient`

Reply:
Replaced the ApiClient with JsonpMapper, thanks @dblock !